vpr/src/place/weighted_median_move_generator.cpp

```diff
@@ -0,0 +1,178 @@
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+
+    auto& place_move_ctx = g_placer_ctx.mutable_move();
+
+    /* Pick a random block to be swapped with another random block.   */
+    ClusterBlockId b_from = pick_from_block();
+
+    if (!b_from) {
+        return e_create_move::ABORT; //No movable block found
+    }
+
+    t_pl_loc from = place_ctx.block_locs[b_from].loc;
+    auto cluster_from_type = cluster_ctx.clb_nlist.block_type(b_from);
+    auto grid_from_type = g_vpr_ctx.device().grid[from.x][from.y].type;
+    VTR_ASSERT(is_tile_compatible(grid_from_type, cluster_from_type));
+
+    /* Calculate the Edge weighted median region */
+    t_pl_loc to;
+
+    t_bb_cost coords;
+    t_bb limit_coords;
+
+    //clear the vectors that saves X & Y coords
+    //reused to save allocation time
+    place_move_ctx.X_coord.clear();
+    place_move_ctx.Y_coord.clear();
+
+    //true if the net is a feedback from the block to itself
+    bool skip_net;
+
+    //iterate over block pins
+    for (ClusterPinId pin_id : cluster_ctx.clb_nlist.block_pins(b_from)) {
+        ClusterNetId net_id = cluster_ctx.clb_nlist.pin_net(pin_id);
+        if (cluster_ctx.clb_nlist.net_is_ignored(net_id))
+            continue;
+        if (int(cluster_ctx.clb_nlist.net_pins(net_id).size()) > placer_opts.place_high_fanout_net)
+            continue;
+        //all net pins are weighted with their own crititcalities
+        get_bb_cost_for_net_excluding_block(net_id, &coords, b_from, pin_id, criticalities, skip_net);
+        if (skip_net)
+            continue;
+
+        place_move_ctx.X_coord.insert(place_move_ctx.X_coord.end(), ceil(coords.xmin.criticality * 10), coords.xmin.edge);
+        place_move_ctx.X_coord.insert(place_move_ctx.X_coord.end(), ceil(coords.xmax.criticality * 10), coords.xmax.edge);
+        place_move_ctx.Y_coord.insert(place_move_ctx.Y_coord.end(), ceil(coords.ymin.criticality * 10), coords.ymin.edge);
+        place_move_ctx.Y_coord.insert(place_move_ctx.Y_coord.end(), ceil(coords.ymax.criticality * 10), coords.ymax.edge);
+    }
+
+    if ((place_move_ctx.X_coord.size() == 0) || (place_move_ctx.Y_coord.size() == 0))
+        return e_create_move::ABORT;
+
+    //calculate the weighted median region
+    std::sort(place_move_ctx.X_coord.begin(), place_move_ctx.X_coord.end());
+    std::sort(place_move_ctx.Y_coord.begin(), place_move_ctx.Y_coord.end());
+
+    if (place_move_ctx.X_coord.size() == 1) {
+        limit_coords.xmin = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2)];
+        limit_coords.xmax = limit_coords.xmin;
+    } else {
+        limit_coords.xmin = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2)];
+        limit_coords.xmax = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2) + 1];
+    }
+
+    if (place_move_ctx.Y_coord.size() == 1) {
+        limit_coords.ymin = place_move_ctx.Y_coord[floor((place_move_ctx.Y_coord.size() - 1) / 2)];
```

Comment:
Same comment as for the x direction: change the vector access for .ymin for [0] or move the .ymin code outside the if.

Reply:
Fixed!